src/dbus.c

```diff
@@ -494,8 +494,16 @@ static void dbus_cb_CloseNotification(
 {
         guint32 id;
         g_variant_get(parameters, "(u)", &id);
-        if (!settings.ignore_dbusclose) 
+        if (settings.ignore_dbusclose) {
+                // Stay commpliant by lying to the sender,  telling him we closed the notification 
+                struct notification *n = dbus_message_to_notification(sender, parameters);
```

Comment:
That's not the right call here. `dbus_message_to_notification` constructs a new notification based on the contents of the message (used for the initial `Notify` call). So what this does is create a new notification and immediately close it.

There isn't a way to retrieve a notification via id currently so it looks like you're going to have to add it to `queues.c`, a simple linear search through the active queues (`waiting`, `displayed`) should be enough.

Reply:
Oh yes… I've jumped on the first matching function without having grok the overall scheme.
Thank you for your insight !

I've reworked the patchset to search the notification in the queues

Note that I'm not so proud of my English and as such, the documentation may also welcome another eye.